src/tests/tracing/runtimeeventsource/NativeRuntimeEventSourceTest.cs

```diff
@@ -31,14 +32,19 @@ static int Main(string[] args)
                     if (OperatingSystem.IsWindows())
                         DoOverlappedIO();
 
-                    // Wait for events.
-                    Thread.Sleep(1000);
-
                     // Generate some GC events.
                     GC.Collect(2, GCCollectionMode.Forced);
 
-                    // Wait for more events.
-                    Thread.Sleep(1000);
+                    Stopwatch sw = Stopwatch.StartNew();
+
+                    while (sw.Elapsed <= TimeSpan.FromMinutes(1))
+                    {
+                        Thread.Sleep(100);
+                        if (listener.SeenProvidersAndEvents.Contains("Microsoft-Windows-DotNETRuntime/EVENTID(65)"))
```

Comment:
Does this mean that if someone adds proper "support" for this event and makes it named this will stop working and the loop will run for 1 minute?

Reply:
Yes, but the support is in the TraceEvent nuget package so it will happen when someone purposefully updates the package, and the test will fail when that happens. It should be fairly obvious what the failure is

Reply:
Actually, almost none of what I said was accurate - I was assuming things. This test uses EventListener and does it all in proc, so it's actually fine and won't break in the future. The eventlistener formats the strings with the provider and event id.